### DISCUSSION TITLE
Entegrasyonların test çalışması

### DISCUSSION THREAD

**Discussion by sekizlipenguen** [mewebstudio/pos#245] [General]
Bu task readme yazan testleri genişlettiğim için açıyorum. Şu ana kadar 3d modeli ile aşağıdakileri test ettim ve çalışmasında problem göremedim. Güncel versiyon ile test Sonuçları(versiyon: 1.4.0)

**ödeme**

- ak bank
- iş bankası
- garanti bankası
- halk bank
- kuveyt turk

**çalıştığı test edilen iptal ve iadeler**

- ak bankası
- iş bankası
- deniz bank
- garanti bankası
- halk bank
- kuveyt turk

**çalıştığı test edilen sipariş durumu kontrolü**

- ak bank
- iş bankası
- deniz bankası
- garanti bankası
- halk bank

**Comment by nuryagdym** [comment]
Merhabalar,
Test ortamda test ediyorsaniz, bazen calismadigi oluyor.

**Comment by nuryagdym** [comment]
Denizbank calisiyorsa ben #57 yi kapatayim.

**Comment by sekizlipenguen** [comment]
Selamlar Deniz bank 3d yöntemini denedim sorunsuz ödeme, iade,iptal işlemleri gerçekleşiyor.

Yukarıda denediklerimin hepsi canlı ortamıdır. Ziraat bankası da öyle canlı ortamda deniyorum beyaz ekranda takılıp kalıyor bir şey eksik mi bilemedim de

**Comment by nuryagdym** [comment]
ziraat ile akbank ayni gateway kullanir. sorunu sadece ziraatta yasiyorsaniz bank ile alakali olabilir.

**Comment by nuryagdym** [comment]
bir de EstPos'un v3 versiyonunu eklemistim.
Ziraat icin onu deneyebilirsiniz. https://github.com/mewebstudio/pos/blob/574bccaf21c3cbf6c2d7886cf0c798a67f8c7f40/config/pos.php#L43

**Comment by sekizlipenguen** [comment]
Selam,

ziraat bankası dönüş sağladı. Api dokümanı iletti sanıyorum ki bizim sistemler baya eskide kalmış. ben url değiştirerek denedim fakat parametrelerde eski olabilir.

```
3D Secure servis adresleri; ++
MPI enrollment adresi : https://mpi.ziraatbank.com.tr/Enrollment.aspx
Sanalpos işlem gönderim servis adresi: https://sanalpos.ziraatbank.com.tr/v4/v3/Vposreq.aspx
Ortak ödeme servis adresleri; ++
Ortak Ödeme Web Servis Adresi (Token Alma Api ile): https://yonetim.ziraatbank.com.tr/v4/api/RegisterTransaction
Ortak Ödeme Web Yönlendirme Adresi: [https://yonetim.ziraatbank.com.tr/v4/SecurePayment?Ptkn={PaymentToken}](https://yonetim.ziraatbank.com.tr/v4/SecurePayment?Ptkn=%7bPaymentToken%7d)
Ortak Ödeme İşlem Sonuç Sorgulama adresi : https://sanalpos.ziraatbank.com.tr/v4/UIWebService/Search.aspx
```

Ziraat bankası ile görüştüm uzun zamandır innova firması ile çalıştıklarını  EstPos yöntemi olmadığını belirttiler. Sanırım biz v4 diye yeni class açıp geliştirme yapmalıyız? Sence nasıl ilerlemeliyiz? 

https://www.innova.com.tr/tr/payflex-vpos-yonetimi 

[PHP_V4.zip](https://github.com/mewebstudio/pos/files/11530984/PHP_V4.zip)

**Comment by nuryagdym** [comment]
merhabalar, ben inceleyeyim biraz.
Dokumanlara yuzeysel bakitigimda Common Payment ifadesini gordum. 
En son vakifbank icin Common Payment gateway entegrasyonu yapmistim, ayni gateway olabilir bu da.

isterseniz `examples/vakifbank-cp` altindaki kodlara bakarak Common Payment Gateway classi kullanarak test etmeyi deneyin.

https://github.com/mewebstudio/pos/blob/master/config/pos.php#L163

Mumkunse bir de epostama ziraatin test API bilgilerini gonderir misiniz ce.nuryagdy@gmail.com

**Comment by sekizlipenguen** [comment]
Selamlar, zıraat için e-posta ilettim. sanırım önce senin incelemen daha iyi olur yapabileceğim bir şey olursa bilgilendirmen yeterli olur, kolay gelsin

**Comment by sekizlipenguen** [comment]
> Selamlar,
> 
> Bu task readme yazan testleri genişlettiğim için açıyorum. Şu ana kadar 3d modeli ile aşağıdakileri test ettim ve çalışmasında problem göremedim.
> 
> **ödeme**
> 
> * ak_bank
> * is_bank
> * garanti_bank
> * qnb_finans_bank
> * yapikredi_bank
> * deniz_bank
> * halk_bank
> * teb bank
> * şeker bank
> 
> **çalıştığı test edilen iptal ve iadeler**
> 
> * ak_bank
> * is_bank
> * garanti_bank
> * qnb_finans_bank
> * yapikredi_bank
> * deniz_bank
> * halk_bank
> * teb bank
> * şeker bank
> 
> **yeni eklenenler**
> 
> * şeker bank - asseco-see sistemini kullandığı için pos.php güncellemesi yaptım, canlı ortamda testler başarılı oldu. pull request kabul edilmesi gerekiyor.
> * * zıraat bankası  - pull request gönderildi.

@nuryagdym  yeni testler ile birlikte konuyu güncelledim. kodlarda güncellemeler yaptım, kütüphaneyi aktif olarak kullandığım için kontrol ve düzenlemelerinden sonra yeni versiyonu çıkmanı bekliyorum.

**Comment by mhmtpacal** [comment]
@sekizlipenguen Merhabalar elinizdeki test hesaplarını ve test kartlarını mail olarak veya buraya gönderebilirmisiniz ?
mail : mhmtpccl@gmail.com

**Comment by sekizlipenguen** [comment]
Selamlar,

@nuryagdym  deniz bank entegrasyonunda ödeme başarılı olmasına rağmen declined dönüşü alıyorum. Bu arada tek tek yeni versiyonda kontrol etmeye başladım.

iş bankası çalışıyor.
ak bank çalışıyor.
deniz bank başarısız diyor ancak çekiyor loglar aşağıda;

   'class' => Mews\Pos\Gateways\InterPos::class,
   
   raw datayı kontrol ettiğimde 3d sonrasın dönüşte şu değerler geliyor.
   
   array:39 [▼ // vendor/mews/pos/src/DataMapper/ResponseDataMapper/InterPosResponseDataMapper.php:337
  "Version" => ""
  "MerchantID" => ""
  "ShopCode" => "gizlendi"
  "TxnStat" => "Y"
  "MD" => "gizlendi"
  "RetCode" => ""
  "RetDet" => ""
  "VenderCode" => ""
  "Eci" => "02"
  "PayerAuthenticationCode" => "gizlendi="
  "PayerTxnId" => ""
  "CavvAlg" => ""
  "PAResVerified" => "True"
  "PAResSyntaxOK" => "True"
  "Expiry" => "****"
  "Pan" => "gizlendi******gizlendi"
  "OrderId" => "gizlendi"
  "PurchAmount" => "1"
  "Exponent" => ""
  "Description" => ""
  "Description2" => ""
  "Currency" => "949"
  "OkUrl" => "gizlendi"
  "FailUrl" => "gizlendi"
  "3DStatus" => "1"
  "AuthCode" => ""
  "HostRefNum" => "hostid"
  "TransId" => ""
  "TRXDATE" => ""
  "CardHolderName" => ""
  "mdStatus" => "1"
  "ProcReturnCode" => ""
  "TxnResult" => ""
  "ErrorMessage" => ""
  "ErrorCode" => ""
  "Response" => ""
  "HASH" => "gizlendi="
  "HASHPARAMS" => "Version:PurchAmount:Exponent:Currency:OkUrl:FailUrl:MD:OrderId:ProcReturnCode:Response:mdStatus:"
  "HASHPARAMSVAL" => "gizlendi"
]

geçici çözüm için şunu yaptım sanıyorum ki şu 2 değer 1 geldiğinde çekim başarılı da demek.

` if ($payment->isSuccess() || ($response['all']['mdStatus'] == 1 && $response['all']['3DStatus'] == 1)) `

**Comment by nuryagdym** [comment]
Merhaba @sekizlipenguen, bunun icin ayri issue acar misin, orda cozelim bu sorunu.
MODEL_3D_SECURE odeme mi kullaniyorsun?
